gui/system/forms.py

```diff
@@ -146,6 +146,19 @@ def clean_path_locked(mp):
             )
 
 
+def check_certificate(certificate):
+    regex = re.compile(r"(-{5}BEGIN[\s\w]+-{5}[^-]+-{5}END[\s\w]+-{5})+", re.M | re.S)
```

Comment:
This regex is still being compiled each call to this function whereas it can just be defined once globally. outside any function or class.